lib/src/logic.dart

```diff
@@ -493,6 +493,9 @@ class Logic {
 
   /// Handles the actual connection of this [Logic] to [other].
   void _connect(Logic other) {
+    if (this == other) {
+      throw InvalidConnectionException(this, other);
+    }
     _unassignable = true;
     _updateWire(other._wire);
```

Comment:
I think we should be able to also/instead check that `this._wire` is the same as `other._wire`.  `Logic`s that are connected without any gates in between should share an underlying `Wire`.  This way we can detect far-off `Logic`s that are actually the same wire connected to itself as well.

Reply:
Yes, it will work. Got the clarity.